plugins/cue_points/base/lib/kCuePointManager.php

```diff
@@ -615,20 +644,118 @@ public static function copyCuePointsFromLiveToVodEntry( $mediaInfo )
 				}
 			}
 		}
-
 		KalturaLog::info("Post processing cuePointIds for live entry [{$liveEntry->getId()}]: " . print_r($processedCuePointIds,true) );
 		if ( count($processedCuePointIds) )
 		{
 			self::postProcessCuePoints( $liveEntry, $processedCuePointIds );
 		}
 	}
-	
+
+	private static function getOffsetForTimestamp($timestamp, $AMFs){
+		KalturaLog::debug('getOffsetForTimestamp ' . $timestamp);
+
+		$minDistanceIndex = self::getClosestAMFIndex($timestamp, $AMFs);
```

Comment:
consider returning amfObject instead of index to avoid access to the array repeatedly.